### DISCUSSION TITLE
Install `mkdocs-material-insiders` with docker leads to an `insufficient_scope` error

### DISCUSSION THREAD

**Discussion by robingenz** [squidfunk/mkdocs-material#5171] [Q&A] [answered]
### Description

I am trying to install mkdocs-material-insiders with Docker as described [here](https://squidfunk.github.io/mkdocs-material/insiders/getting-started/#with-docker).
I followed the steps exactly and after creating the GitHub release I get the following error message in the GitHub Actions workflow in the `Publish Docker image` step:

```
...
#37 exporting attestation manifest sha256:be81eefd5c44acc495e20f19a82a05d30f229cf128b02c673ddc9c9296cd487b 0.0s done
#37 ...

#38 [auth] robingenz/mkdocs-material-insiders:pull,push token for registry-1.docker.io
#38 DONE 0.0s

#37 exporting to image
#37 exporting manifest list sha256:db83a971881c921604296ee13b74c81098ae2c3667e1a1af0a379b1717d49ad3 done
#37 pushing layers
#37 ...

#39 [auth] robingenz/mkdocs-material-insiders:pull,push token for registry-1.docker.io
#39 DONE 0.0s

#40 [auth] robingenz/mkdocs-material-insiders:pull,push token for registry-1.docker.io
#40 DONE 0.0s

#41 [auth] robingenz/mkdocs-material-insiders:pull,push token for registry-1.docker.io
#41 DONE 0.0s

#42 [auth] robingenz/mkdocs-material-insiders:pull,push token for registry-1.docker.io
#42 DONE 0.0s

#37 exporting to image
#37 pushing layers 0.3s done
#37 ERROR: failed to push robingenz/mkdocs-material-insiders:9.1.1-insiders-4.32.2: server message: insufficient_scope: authorization failed
------
 > exporting to image:
------
ERROR: failed to solve: failed to push robingenz/mkdocs-material-insiders:9.1.1-insiders-4.32.2: server message: insufficient_scope: authorization failed
Error: buildx failed with: ERROR: failed to solve: failed to push robingenz/mkdocs-material-insiders:9.1.1-insiders-4.32.2: server message: insufficient_scope: authorization failed
```

### Related links

- [Installation with docker](https://squidfunk.github.io/mkdocs-material/insiders/getting-started/#with-docker)

### Before submitting

- [X] I have read and followed the [documentation issue reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-docs-issue/).
- [X] I have attached the links to the described sections of [the documentation](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-docs-issue/#related-links)

**Comment by squidfunk** [comment, marked as answer]
Thanks for reporting. The error message is actually pretty clear – the access token has insufficient scopes. More specifically, the `write:packages` scope seems to be missing. Please note that you need to use a specific token, called `GHCR_TOKEN` – just in case you missed it. Additionally, please make sure that you really [followed the steps outlined in the documentation](https://squidfunk.github.io/mkdocs-material/insiders/getting-started/#with-docker), because this is what we all use successfully to build the Docker image. Specifically:

- Create a new personal access token ([why?](https://squidfunk.github.io/mkdocs-material/insiders/getting-started/#fn:4))
  - Go to https://github.com/settings/tokens
  - Click on [Generate a new token](https://github.com/settings/tokens/new)
  - Enter a name and select the [`write:packages`](https://docs.github.com/en/developers/apps/scopes-for-oauth-apps#available-scopes) scope
  - Generate the token and store it in a safe place
- Add a [GitHub Actions secret](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository) on your fork
  - Set the name to `GHCR_TOKEN`
  - Set the value to the personal access token created in the previous step

Converting to discussion, since this I'm confident that this is not an issue with our documentation.

**Reply by squidfunk** [reply]
All this is in case you use the Docker [build and release workflow provided by us](https://github.com/squidfunk/mkdocs-material-insiders/blob/master/.github/workflows/build.yml), which uses the GitHub Container Registry. Note that if you use another registry like DockerHub, you're more or less on your own and might need to adjust those settings.

Thus, we recommend just using GHCR.

**Reply by robingenz** [reply]
Thank you for your fast response!

> More specifically, the `write:packages` scope seems to be missing.

This is already set:

![grafik](https://user-images.githubusercontent.com/13857929/223947036-55ac3b7e-7d1c-43ad-b110-75d0ee641437.png)

> Please note that you need to use a specific token, called GHCR_TOKEN – just in case you missed it.

I also did that:

![grafik](https://user-images.githubusercontent.com/13857929/223947338-6fcf855a-7ccc-470e-887b-d46d6ef0a4f6.png)

It seems to be trying to push the image to DockerHub or logging into DockerHub instead of GitHub Packages because of this log entry:

```
...
#42 [auth] robingenz/mkdocs-material-insiders:pull,push token for registry-1.docker.io
#42 DONE 0.0s
...
```

Could this possibly be the problem? However, I have not changed anything in the workflow. It is the same as in the fork.

**Reply by squidfunk** [reply]
Please try to sync the Insiders repository, we recently fixed an issue with the build script.
https://github.com/squidfunk/mkdocs-material-insiders/commit/68510cb1cedff4c15b09939a308bb84437c3da27

**Reply by robingenz** [reply]
Great, that works. My mistake was to use the latest release and not test the current master version. Thanks for your help!

**Reply by squidfunk** [reply]
My fault to not release it immediately. Just released 4.23.3 with the latest workflow.
